src/modules/languages/javascript.nix

```diff
@@ -218,7 +218,7 @@ in
     ]
     ++ lib.optional cfg.npm.enable (cfg.npm.package)
     ++ lib.optional cfg.pnpm.enable (cfg.pnpm.package)
-    ++ lib.optional cfg.yarn.enable (cfg.yarn.package)
+    ++ lib.optional cfg.yarn.enable (cfg.yarn.package.override { nodejs = cfg.package; })
```

Comment:
do we also need this for pnpm?

Reply:
Yea pnpm has the same problem but this override only works because yarn package supports it. Pnpm is a nodePackage which sadly doesn't support this (same problem if you would use nodePackage.yarn instead of yarn)

Reply:
Only way I found to fix that was using overlay to fix all nodejs to specific version

Reply:
Underlying nix issue: https://github.com/NixOS/nixpkgs/issues/145634

Reply:
Something like this also works: `cfg.package.pkgs.pnpm`. This installs pnpm with the correct node version, but not sure if all node pkgs support this (tested with `pkgs.nodejs_21;`), will do some more testing.

Reply:
Doesn't work with nodejs-slim pkg so not sure what the best way forward is here. Supporting overlays seems like the only way to support all different pkgs.

Reply:
Commented workarounds in original issue aswel so people can still use this until a proper fix is ready. Do you reckon moving away from slim nodejs is a good solution?